File: src/main/java/org/pl/parser/ast/NumberNode.java
package org.pl.parser.ast;


import java.math.BigDecimal;

/**
 * Single float value.
 */
public class NumberNode implements INode {
    public final BigDecimal value;

    public NumberNode(BigDecimal value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "NumberNode{val=" + value + '}';
    }
}

File: src/main/kotlin/org/leaflang/parser/ast/value/NumberNode.kt
package org.leaflang.parser.ast.value

import org.leaflang.parser.ast.INode
import org.leaflang.parser.utils.NodePosition
import java.math.BigDecimal

/**
 * Single float value.
 */
data class NumberNode(override val position: NodePosition,
                      val value: BigDecimal) : INode